tests/dialects/test_bigquery.py

```diff
@@ -18,6 +18,17 @@ class TestBigQuery(Validator):
     maxDiff = None
 
     def test_bigquery(self):
+        self.validate_all(
+            "SELECT STRUCT(1, 2, 3), STRUCT(), STRUCT('abc'), STRUCT(1, t.str_col), STRUCT(1 as a, 'abc' AS b), STRUCT(str_col AS abc)",
+            write={
+                "bigquery": "SELECT STRUCT(1, 2, 3), STRUCT(), STRUCT('abc'), STRUCT(1, t.str_col), STRUCT(1 AS a, 'abc' AS b), STRUCT(str_col AS abc)",
+                "duckdb": "SELECT {'_0': 1, '_1': 2, '_2': 3}, {}, {'_0': 'abc'}, {'_0': 1, '_1': t.str_col}, {'a': 1, 'b': 'abc'}, {'abc': str_col}",
+                "spark": "SELECT STRUCT(1, 2, 3), STRUCT(), STRUCT('abc'), STRUCT(1, t.str_col), STRUCT(1 AS a, 'abc' AS b), STRUCT(str_col AS abc)",
+                "snowflake": "SELECT OBJECT_CONSTRUCT('', 1, '', 2, '', 3), OBJECT_CONSTRUCT(), OBJECT_CONSTRUCT('', abc), OBJECT_CONSTRUCT('', 1, '', str_col), OBJECT_CONSTRUCT('a', 1, 'b', 'abc'), OBJECT_CONSTRUCT('abc', str_col)",
+                # struct is a fallback without type inference
+                "trino": "SELECT ROW(1, 2, 3), ROW(), ROW('abc'), ROW(1, t.str_col), CAST(ROW(1, 'abc') AS ROW(a INTEGER, b VARCHAR)), STRUCT(str_col AS abc)",
+            },
+        )
```

Comment:
Note that:

- `STRUCT()` is generally not supported I think in these dialects, only Snowflake allowed `OBJECT_CONSTRUCT()`
- This is invalid Snowflake `OBJECT_CONSTRUCT('', 1, '', 2, '', 3)` due to duplicate keys

Not sure if we wanna do something about this, just observing.